messaging/src/main/java/org/axonframework/configuration/Component.java

```diff
@@ -147,6 +149,19 @@ public boolean isInitialized() {
         return instance != null;
     }
 
+    @Override
+    public void describeTo(@Nonnull ComponentDescriptor descriptor) {
+        descriptor.describeProperty("identifier", identifier.toString());
+        if (isInitialized()) {
+            descriptor.describeProperty("component", instance);
+            descriptor.describeProperty("initialized", true);
+        } else {
+            descriptor.describeProperty("factory", factory);
+            descriptor.describeProperty("decorators", decorators);
+            descriptor.describeProperty("initialized", false);
+        }
```

Comment:
I'm not sure about that - what should be our convention for such cases. 

Should we skip a property or better pass null or some default value like "undefined" / "not-initialized'? Then we make it explicit that it's not ready yet, but we didn't forgot about that. 

Then it may be a decision of the `ComponentDescriptor` if you'd like to include null/"not-initialized" properties in the result. Then it may work similar like in JSON serializers - you may include null properties or just skip them - depends on the configuration.

Reply:
I don't understand fully what do you mean by this question:
`Does it even make sense to keep it as an implementation detail if you want to enforce all properties?` ? 
But yes, I will move the responsibility what to do with null properties to the `ComponentDescriptor`.

Reply:
I gave it a thought and now I'm not sure... maybe what you have done it's better. Because it's like a state machine right now - we have initialized/ not-initialized with just different properties. My thinking was too "property-centric".

Reply:
Ah, sorry for causing doubt here, @MateuszNaKodach! 
Based on your question, I was wondering whether we can even enforce a standard easily on the API.

Having also given it some thought, I don't think we can. Differently put, the implementation of the `describeTo` operation is to the discretion of the person implementing a `DescribableComponent`.

Nonetheless, I do think that as a rule of thumb, it would be good to state in the Javadoc of `describeTo` we should describe all properties that are present.
However, there will always be outliers, and I think the configuration is one of them.